thoth/investigator/investigate_unresolved_package.py

```diff
@@ -200,7 +199,8 @@ def _check_package_version(package_name: str, package_version: Optional[str], in
         source = Source(index_url)
 
         try:
-            versions = source.get_package_versions(package_name)
+            # Get sorted versions (latest first -> latest = versions[0])
+            versions = source.get_sorted_package_versions(package_name)
```

Comment:
It could be done here all at once `map(str, source.get_sorted_package_versions(package_name))`